Model/Tax/Calculation/AdvancedCrossBorderCalculator.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * Copyright © 2020 FireGento e.V.
+ * See LICENSE.md bundled with this module for license details.
+ */
+
+declare(strict_types=1);
+
+namespace FireGento\MageSetup\Model\Tax\Calculation;
+
+use FireGento\MageSetup\Model\System\Config as SystemConfig;
+use Magento\Framework\DataObject;
+use Magento\Tax\Api\Data\AppliedTaxInterfaceFactory;
+use Magento\Tax\Api\Data\AppliedTaxRateInterfaceFactory;
+use Magento\Tax\Api\Data\TaxDetailsItemInterfaceFactory;
+use Magento\Tax\Api\TaxClassManagementInterface;
+use Magento\Tax\Model\Calculation;
+use Magento\Tax\Model\Config;
+
+/**
+ * Trait AdvancedCrossBorderCalculator
+ *
+ * @package FireGento\MageSetup\Model\Tax\Calculation
+ */
+trait AdvancedCrossBorderCalculator
```

Comment:
I am not a big fan of traits, but this seems to be a proper use case indeed :-)

Reply:
as I explained in commit message, other approaches here won't work or will cause code duplication.

more generally I have a strong feeling that magneto should support that out of the box or at least expose interfaces for easier override of this piece.